Draft: Create content for "Choosing a Laptop" (#20)
EssamWisam/cmp-docs: public/department/General/Recently/Laptop.md

# 💻 Choosing a Laptop

Generally, the minimum needed specs for a laptop are not that high. Let's begin by quickly going over the main parts of the laptop. Each part will get its own brief section. It's advised that you at least skim through this whole document once.

Note that unlike desktop PCs where you can mostly mix and match parts, laptops come mostly as a whole package. You're usually able to upgrade the storage/RAM, but that's it. So you're stuck with whatever choose until you upgrade to a new laptop. Make sure to plan a little ahead and be prepared to strech your budget if you plan on keeping your laptop for long (5+ years).

## Picking a laptop in a nutshell

> Set a budget range, filter out all laptops that don't fit your range or aren't in stock, filter out laptops that dont fit your usage, the watch as many reviews as you can for each laptop so you get to know the pros and cons for each. You can then settle on one and buy it. Keep in mind that you can buy from Egypt for local warranty (for what it's worth), or from outside if you want to save money.

Reviewer: ```suggestion
Set a budget range, filter out all laptops that don't fit your range or aren't in stock, filter out laptops that dont fit your usage, then watch as many reviews as you can for each laptop so you get to know the pros and cons for each. You can then settle on one and buy it. Keep in mind that you can buy from Egypt for local warranty (for what it's worth), or from abroad if you want to save money.
```

## Laptop types
Laptops generally fall under three broad types: thin and light, gaming, and workstation laptops.

### Thin and Lights
This type of laptop sacrifices performance and sometimes upgradability for the sake of being the most portable with the smallest chargers and longest battery life. You usually will have to pay more for similar specs to other types while receiving lower performance due to inferior cooling or power budget.

### Gaming 
Performance at the cost of everything else. These are heavy and hot, but they sure (usually) run fast. They also don't last long, be prepared to carry around a heavy power brick if you intend on doing more than one hour of work unplugged. You also pay a price premium for the GPU. So make sure you're going to use it before settling on one.

### Workstations
Sometimes even bulkier than gaming laptops, these laptops are usually bought used as they provide good-ish performance for cheap. They usually have higher end but older parts. Since they are usually bought used, they can hide nasty surprises. Make sure to test them well before buying.

## Laptop parts
This section dissuses the main parts that you should compare laptops over.

### CPU

> The brains of the laptop. This will contribute greatly to how smooth and responsive your experience is with the laptop. These are usually called "Intel Core i7 9750H",  "AMD Ryzen 5 5500U", etc.. Let's begin by breaking down the naming convention.

Reviewer: ```suggestion
The brains of the laptop. This will contribute greatly to how smooth and responsive your experience is with the laptop. These are usually called "Intel Core i7 9750H",  "AMD Ryzen 5 5500U", etc.. Let's begin by breaking down the naming convention:
```

> #### Naming Convention

Reviewer: ```suggestion

```
By removing this (the previous sentence has a colon anyway), you get rid of all of the following h5 titles (make them h4 instead) as h5 renders poorly in the website.

##### Manufacturer

> (Intel/AMD): These are the biggest two names in laptop processors at the time of writing (if we ignore apple for a minute), they're usually about equal, we'll discuss them a bit later on. 

Reviewer: ```suggestion
(Intel/AMD): These are the biggest two names in laptop processors at the time of writing (if we ignore Apple for a minute), they're usually about equal, we'll discuss them a bit later on. 
```

If your budget is tight and the laptop comes with only one stick, you can later add another stick (check the laptop spec sheet or a teardown for this, keep in mind the model) if needed. If your laptop comes with one slot or two filled slots, you can always sell a stick or two off and buy a better one.

Note that some laptops (mostly thin and light ones or budget ones) come with soldered ram, *YOU WILL NOT BE ABLE TO UPGRADE*, so make sure to overprovision.

You'll be running a lot of things that are ram hungry for projects and labs, so the bare minimum you should aim for is 8GB (terrible idea by the way), 16GB+ is strongly preferred if possible. 

### Screen

> #### Resolution

Reviewer: ```suggestion

#### Resolution
```

> Displays contents from the laptop. You'll be staring at this for hours every day. It's advised to at least aim for 1920x1080 for the resolution. You can live with lower resolutions, but they won't be super pleasant.

Reviewer: ```suggestion
Displays contents from the laptop. You'll be staring at this for hours every day. It's advised to at least aim for 1920x1080 for the resolution. You can live with lower resolutions, but staring at them for a long time won't feel as good to your eyes.
```

You usually cannot compare laptops by these specs alone, it's advised you pick two different laptop chassis with similar specs and watch/read reviews and compare thermals (averge, max temperature as well as CPU/GPU power).

### Battery
Controls how long/hard you can run your laptop without the charger. Larger capacity (Watt-hours or "wh") is generally better, but might lead to a heavier weighing laptop.

### Chasis
The skeleton of your laptop. Mainly determines how rugged/premium your laptop feels. The more money you spend, the more rigid the chassis will be, with more premium materials (aluminum vs plastic). Screen hinges are also a notable part. Some laptops are known to have widespread hinge failures, so make sure to look those up.

### Inputs
Mainly the keyboard and mouse. You can't really eye this, so you either have to go and test a real-life laptop in some store, or watch reviews. You can always use an external mouse/keyboard, but keep in mind that you'll have to carry it with you, and it will take up USB ports.

### Storage Drives

> Controls how much stuff you can download (pirate) from the internet. Storage drives have two types: NVMe SSDs and SATA HDDs (ignoring SATA SSDs since they aren't super popular for laptops.). The former is the faster, lighter, and (of course) more expensive option, while the latter allows you to get more storage capacity for cheaper. Some laptops come with only the former, some come only with the latter (cheaper ones), and some come with both. The author highly recommends at least an NVMe SSD for the OS (Windows or Linux). You can still live with a SATA HDD, but it will be SLOW (as in five minutes to boot slow vs 30 or less on an NVMe SSD).

Reviewer: ```suggestion
Controls how much stuff you can download (pirate) from the internet. Storage drives have two types: NVMe SSDs and SATA HDDs (ignoring SATA SSDs since they aren't super popular for laptops.). The former is the faster, lighter, and (of course) more expensive option, while the latter allows you to get more storage capacity for cheaper. Some laptops come with only the former, some come only with the latter (cheaper ones), and some come with both. The author highly recommends at least an NVMe SSD for the OS (Windows or Linux). You can still live with a SATA HDD, but it will be SLOW (as in five minutes to boot slow vs 30 or less seconds on an NVMe SSD).
```

> You can always expand your storage (check the spec sheet for available ports), or use an external drive. Most new laptops come with at least 256GB of NVMe storage or 1TB of HDD storage (or both). 256GB should be the bare minimum you go withj. At this size, you'll have to a lot more maintenance work like deleting temporary files and scanning for large files frequently. Also, forget doing any long-term storage (images, movies, games, course materials, etc... Higher capacities (512GB+) are strongly adivsed.

Reviewer: ```suggestion
You can always expand your storage (check the spec sheet for available ports), or use an external drive. Most new laptops come with at least 256GB of NVMe storage or 1TB of HDD storage (or both). 256GB should be the bare minimum you go with. At this size, you'll have to a lot more maintenance work like deleting temporary files and scanning for large files frequently. Also, forget doing any long-term storage (images, movies, games, course materials, etc... Higher capacities for the NVMe (512GB+) are strongly recommended.
```

Generally, the minimum needed specs for a laptop are not that high. Let's begin by quickly going over the main parts of the laptop. Each part will get its own brief section. It's advised that you at least skim through this whole document once.

Note that unlike desktop PCs where you can mostly mix and match parts, laptops come mostly as a whole package. You're usually able to upgrade the storage/RAM, but that's it. So you're stuck with whatever choose until you upgrade to a new laptop. Make sure to plan a little ahead and be prepared to strech your budget if you plan on keeping your laptop for long (5+ years).

## Picking a laptop in a nutshell
Set a budget range, filter out all laptops that don't fit your range or aren't in stock, filter out laptops that dont fit your usage, the watch as many reviews as you can for each laptop so you get to know the pros and cons for each. You can then settle on one and buy it. Keep in mind that you can buy from Egypt for local warranty (for what it's worth), or from outside if you want to save money.

## Laptop types
Laptops generally fall under three broad types: thin and light, gaming, and workstation laptops.

### Thin and Lights

> This type of laptop sacrifices performance and sometimes upgradability for the sake of being the most portable. You usually will have to pay more for similar specs to other types while receiving lower performance. Pick this if your commute is long and your back/shoulders are weak, and you don't really care about performance.

Reviewer: I would suggest that we use a more cautious language while presenting facts. So this could be made into 
> This type of laptop often sacrifices/may sacrifice performance and sometimes...
You can also mention a proof to further convince the reader, for instance `xyz` laptop has about similar specs to `abc` laptop while being about `140%` of the price in 2023.

Reviewer: > Pick this if your commute is long and your back/shoulders are weak

Do you mean?

> Pick this if you don't want to break your back by buying a gaming laptop with a 1KG GPU and related stuff that you will almost never use.

Kidding of course, both are extreme but the point is you should expose this as a pro of this type of laptops (and not a con in whoever is buying it) since you are an unbiased liberal writer.

Reviewer: > and you don't really care about performance.

WoW. You destroyed them here. I think liberal unbiased Tarek would rather write:

> and you are willing to raise the budget a little for decent performance.

Author: > I would suggest that we use a more cautious language while presenting facts. So this could be made into
> 
> > This type of laptop often sacrifices/may sacrifice performance and sometimes...
> 
> You can also mention a proof to further convince the reader, for instance `xyz` laptop has about similar specs to `abc` laptop while being about `140%` of the price in 2023.

- I guess I was a bit too harsh on the thin and light crowd, will reword. 
- For the comparison of laptops, the prices fluctuate and depend on the shop you're buying from, so I suppose the best option would be letting the reader form their own opinion on the matter. 
- Pros and cons would probably be more concise, but lists are still broken so they look kinda ugly. Might be a good improvement for the future though!

### Gaming 

> Performance at the cost of everything else. These are heavy and hot, but they sure (usually) run fast. Be prepared to carry around a heavy power brick if you intend on doing more than one hour of work unplugged.

Reviewer: Again, a pros versus cons here would be really helpful. I find the heavy weight and extra bucks paid for the GPU, which most don't use to be a serious con, eh?

Author: Noted!

### Workstations
Even bulkier than gaming laptops, these laptops are usually bought used as they provide good-ish performance for cheap. They usually have higher end but older parts.

## Laptop parts
This section dissuses the main parts that you should compare laptops over.

### CPU

> The brains of the laptop. This will contribute greatly to how smooth and responsive your experience is with the laptop. These are usually called "Intel Core i7 9750H",  "AMD Ryzen 5 5500U", etc.. Let's begin by breaking down the naming convention.

Reviewer: A small introduction into what does it mean to have higher clock speed (more lines of code per second) and hence, faster startup, processing of items, etc. and likewise, for cores and any other main specs of the CPU you like.

Author: Done! Didn't go really in depth as I wanted since its quite hard to boil down all the nuance.  Eagerly awaiting your feedback.

##### Series
This is the "H" or "U" in intel processors, or "HX"/"HS" or "U" in AMD processors. In short, "H"/"HX"/"HS" processors perform better, but have worse battery life, consume more power, and require more cooling than their "U" counterparts.

#### Comparisons
One must note that one processor can perform very differently with different power consumption limits and different cooling systems, this must be taken in mind when comparing two laptops with the same processor but vastly different cooling systems or power limits.

Another thing that must be taken in mind is that inter-series comparisons must be understood correctly. You cannot compare an F1 car to a gas efficient consumer car. Each is designed for a specific purpose and this must be kept in mind. You use one when you want to go FAST, and the other if you like your rides to be comfortable. This also applies to inter-generational comparisons, it's expected that the more recent version of a CPU _should_ be better. 

### RAM

> Mainly contributes to how many applications you can run in parallel. Generally, the higher the ram capacity and frequency are, the better it is. So for example, an 8GB stick of ram is usually better than a 4GB stick of ram. And similarly, a "3200MHz" stick of ram is also better than a "2666MHz" stick of ram. Another dimension to consider is the number of sticks coming with the laptop. Usually, having two sticks installed is better than one for performance, even if the total capacity is the same. So 2x8GB is usually better than 1x16GB. Capacity is more important than frequency here.

Reviewer: It would be nice to emphasize that this is among the most important specifications because an average dev. in our department has 100 chrome tabs open, 20 VSCode tabs open, a VM running in the background and sometimes Microsoft office. This is just an optional recommendation.

If you like, can also put true minimum bounds like that it's a terrible idea to go below 8GB and that it's highly recommended, for someone that wants to keep their laptop as is for long to go beyond 16GB. This would be hard to do for the processor, eh?

Author: Good idea, but I doubt most first year students know VSCode or what a VM is, so perhaps it'd be a good idea to omit the details.

Reviewer: Can be easily swapped out for "code editor" and "virtual machine". I agree that it's not necessary to justify anyway. So you're good.

If your budget is tight and the laptop comes with only one stick, you can later add another stick (check the laptop spec sheet or a teardown for this, keep in mind the model) if needed. If your laptop comes with one slot or two filled slots, you can always sell a stick or two off and buy a better one.

Note that some laptops (mostly thin and light ones or budget ones) come with soldered ram, *YOU WILL NOT BE ABLE TO UPGRADE*, so make sure to overprovision.

> ### Screen

Reviewer: Maybe write below here that this is really an optional section for this that care about the display so someone doesn't feel overwhelmed that they have to read this.

Author: Perhaps it'd be a good idea to ask for feedback from first year students first? I don't want to accidentally make someone half blind because they bought a laptop with a bad screen :P

Reviewer: Fair enough. You know I highly value good screens. I was really referring to cooling, chasis, ports, inputs. These aren't high priority specs.

I agree anyway not to mention "advanced" or anything but just to order the specs by their importance (e.g., CPU, RAM, Storage, GPU, Screen, then the rest)

You usually cannot compare laptops by these specs alone, it's advised you pick two different laptop chassis with similar specs and watch/read reviews and compare thermals (averge, max temperature as well as CPU/GPU power).

### Battery
Controls how long/hard you can run your laptop without the charger. Larger capacity (Watt-hours or "wh") is generally better, but might lead to a heavier weighing laptop.

### Chasis
The skeleton of your laptop. Mainly determines how rugged/premium your laptop feels. The more money you spend, the more rigid the chassis will be, with more premium materials (aluminum vs plastic). Screen hinges are also a notable part. Some laptops are known to have widespread hinge failures, so make sure to look those up.

### Inputs
Mainly the keyboard and mouse. You can't really eye this, so you either have to go and test a real-life laptop in some store, or watch reviews. You can always use an external mouse/keyboard, but keep in mind that you'll have to carry it with you, and it will take up USB ports.

> ### Storage Drives

Reviewer: I think you can divide this section into a beginner and advanced part (or similar) and beginner should only mention things like RAM, CPU and GPU; meanwhile, advanced goes into all the other stuff (chasis, inputs, etc.)

Author: Honestly, I don't think this is quite "advanced". It's more of "stuff that people don't care about until they do", so I think everyone should at least take a glance at everything.

Reviewer: Okay. Do you think we can reorder them around then? Thinking to promote Storage and GPU then speak about the rest (Screen, Ports, etc.) For the GPU, can remove "Optional" from the title and write it as a sentence.

> Controls how much stuff you can download (pirate) from the internet. Storage drives have two types: NVMe SSDs and SATA HDDs (ignoring SATA SSDs since they aren't super popular for laptops.). The former is the faster, lighter, and (of course) more expensive option, while the latter allows you to get more storage capacity for cheaper. Some laptops come with only the former, some come only with the latter (cheaper ones), and some come with both. The author highly recommends at least an NVMe SSD for the OS (Windows or Linux). You can still live with a SATA HDD, but it will be SLOW (as in five minutes to boot slow vs 30 or less on an NVMe SSD).

Reviewer: Could we estimate here what is a minimum storage size to be able to live with the large software we typically need to download in college? Along with a recommended storage size?

Author: Good idea!

You can always expand your storage (check the spec sheet for available ports), or use an external drive.

### Ports
Allows you to connect mice, keyboards, USBs, external harddrives, screens, etc.. to your laptop. Usually the more, the merrier. Mostly controlled by the laptop chassis. The author recommends at least 2 USB-A ports (the big rectangular ones), a USB-C port (the small, round ones), and an HDMI port as a minimum. Other nice to have ports include: a headphone port, an ethernet port, more USB/USB-C ports.

### (Optional) GPU

> Mostly matters for gamers. For machine/deep learning aficionados, you'll probably use some cloud platform to train your models. 

Reviewer: > Mostly matters for gamers. For machine/deep learning aficionados, you'll probably use some cloud platform to train your models. 

Crossed my mind that we can write:
> Mostly matters for gamers and those interested in artificial intelligence (machine learning and deep learning). The latter type of people usually have another preferred option such as training models on the cloud but some of them prefer using their own GPUs offline as it gives more flexibility.

Author: Hope you don't mind me "borrowing" this :)
